File: library/src/main/java/com/goplay/ads/helper/JsonPullerTask.java
/*
 * Created by GoPlay Games.
 * @goplay.mobilegames
 * Copyright (c) 2018.
 */

package com.goplay.ads.helper;

import android.os.AsyncTask;
import android.util.Log;

public class JsonPullerTask extends AsyncTask<String, String, String> {
    private final String jsonUrl;
    private final JsonPullerListener listener;

    public JsonPullerTask(String url, JsonPullerListener jsonPullerListener) {
        this.jsonUrl = url;
        this.listener = jsonPullerListener;
    }

    @Override
    protected String doInBackground(String... p1) {
        return GoPlayAdsHelper.parseJsonObject(jsonUrl);
    }

    @Override
    protected void onPostExecute(String result) {
        listener.onPostExecute(result);
        Log.d("Response", result);
    }

    public interface JsonPullerListener{
        void onPostExecute(String result);
    }
}

File: library/src/main/java/com/goplay/ads/helper/JsonPullerTask.kt
package com.goplay.ads.helper

import android.os.AsyncTask
import android.util.Log
import androidx.annotation.RestrictTo

@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
class JsonPullerTask(private val jsonUrl: String, private val listener: JsonPullerListener) : AsyncTask<String, String, String>() {

    override fun doInBackground(vararg p1: String): String {
        return GoPlayAdsHelper.parseJsonObject(jsonUrl)
    }

    override fun onPostExecute(result: String) {
        listener.onPostExecute(result)
        Log.d("Response", result)
    }

    interface JsonPullerListener {
        fun onPostExecute(result: String)
    }
}
